srv/src/main/java/my/bookshop/handlers/HierarchyHandler.java

```diff
@@ -225,13 +226,13 @@ private class Sorter implements Comparator<GenreHierarchy> {
 
         @Override
         public int compare(GenreHierarchy gh1, GenreHierarchy gh2) {
-            Stack<String> path1 = getPath(gh1);
-            Stack<String> path2 = getPath(gh2);
+            Deque<String> path1 = getPath(gh1);
+            Deque<String> path2 = getPath(gh2);
             int res = 0;
 
             while (!path1.isEmpty() && !path2.isEmpty()) {
-                String last1 = path1.pop();
-                String last2 = path2.pop();
+                String last1 = path1.removeFirst();
+                String last2 = path2.removeFirst();
```

Comment:
```suggestion
                String last1 = path1.pop();
                String last2 = path2.pop();
```

?

Reply:
The changes were made according to the documentation https://docs.oracle.com/javase/8/docs/api/java/util/Deque.html
`push(e) -> addFirst(e)`
`pop() -> removeFirst()`